MewsSystems/mews-flutter
# fix: [RND-178412] OptimusAvatar offline mode throwing exception

RND-178412

#### Summary

- `OptimusAvatar` was throwing an exception when the load was unsuccessful. Now it won't (the error is handled, but now there is no unnecessary noise in the form of an exception in the log).
- I've replaced `CircleAvatar` with `FadeInImage` to add more animation when the image is loaded. Looks better. Before that, it would just pop in all at once.
- I've borrowed a `transparentImage` from the [transparent_image](https://pub.dev/packages/transparent_image), which he borrowed from Dart sources itself(it was private), so I guess we're even.

#### Testing steps

1. Open `Avatar` Story
2. Check is all avatars are displayed correctly.
3. Intentionally set an URL to fail and check if `OptimusAvatar` is set correctly.

#### Follow-up issues

None.

#### Check during review

- Verify against YouTrack issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).
